presto-main/src/test/java/com/facebook/presto/operator/TestPartitionedOutputOperator.java

```diff
@@ -0,0 +1,340 @@
+        }
+        return new Page(positions.size(), blocks);
+    }
+
+    private Page createSimplePage()
+    {
+        return buildPage(TYPES, 1000, false, false, false, false, Optional.empty());
+    }
+
+    private Page createPageWithDictionary()
+    {
+        return buildPage(TYPES, 1000, false, false, false, false, Optional.of("D"));
+    }
+
+    private Page createPageWithRunLength()
+    {
+        return buildPage(TYPES, 1000, false, false, false, false, Optional.of("R"));
+    }
+
+    private DriverContext createDriverContext()
+    {
+        return TestingTaskContext.builder(EXECUTOR, SCHEDULER, TEST_SESSION)
+                .setMemoryPoolSize(MAX_MEMORY)
+                .setQueryMaxTotalMemory(MAX_MEMORY)
+                .build()
+                .addPipelineContext(0, true, true, false)
+                .addDriverContext();
+    }
+
+    private PartitionedOutputBuffer createPartitionedBuffer(OutputBuffers buffers, DataSize dataSize)
+    {
+        return new PartitionedOutputBuffer(
+                "task-instance-id",
+                new StateMachine<>("bufferState", SCHEDULER, OPEN, TERMINAL_BUFFER_STATES),
+                buffers,
+                dataSize,
+                () -> new SimpleLocalMemoryContext(newSimpleAggregatedMemoryContext(), "test"),
+                SCHEDULER);
+    }
+
+    private Page buildPage(
+            List<Type> types,
+            int positionCount,
+            boolean addPreComputedHashBlockAtBegin,
+            boolean addNullBlockAtEnd,
+            boolean allowNulls,
+            boolean useBlockView,
+            Optional<String> wrappings)
+    {
+        int channelCount = types.size();
+        int preComputedChannelCount = (addPreComputedHashBlockAtBegin ? 1 : 0);
+        int nullChannelCount = (addNullBlockAtEnd ? 1 : 0);
+
+        Block[] blocks = new Block[channelCount + preComputedChannelCount + nullChannelCount];
+
+        if (addPreComputedHashBlockAtBegin) {
+            blocks[0] = buildBigintBlock(positionCount, false);
+        }
+
+        for (int i = 0; i < channelCount; i++) {
+            blocks[i + preComputedChannelCount] = buildBlockWithType(types.get(i), positionCount, allowNulls, useBlockView, wrappings);
+        }
+
+        if (addNullBlockAtEnd) {
+            blocks[channelCount + preComputedChannelCount] = buildNullBlock(BIGINT, positionCount);
+        }
+
+        return new Page(positionCount, blocks);
+    }
+
+    Block buildBigintBlock(int positionCount, boolean allowNulls)
+    {
+        return createLongsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextLong())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildBlockWithType(Type type, int positionCount, boolean allowNulls, boolean isView, Optional<String> wrappings)
```

Comment:
buildBlockWithType was created in https://github.com/prestodb/presto/pull/13183 mainly for nested types. Here we don't need to use it because we only need to test on BIGINT and create LongArrayBlock.

Reply:
done